### ISSUE TITLE
"ignore_index" argment for focal loss and dice loss

### ISSUE THREAD

**Issue by legendongary** [kornia/kornia#839]
When I try to train for segmentation on VOC 2012 with Dice loss, I always run into cuda illegal error and I finally found that this may be caused by out-of-range indexes (255 out of 20 classes). Since pytorch CrossEntropyLoss has an argment "ignore_index" to deal with this issue, I wonder if there is a proper way to deal with uncertain indexes?
